src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java

```diff
@@ -112,16 +122,31 @@ UserSubscription collectSubscriptions(List<Subscription> subscriptions) {
     public void collectSubscribers(Artefact artefact) {
         List<Subscription> subscriptionList = new ArrayList<>(querySubscriptionValue(
             SearchType.COURT_ID.name(), artefact.getCourtId()));
-        artefact.getSearch().get("cases").forEach(object -> subscriptionList.addAll(extractSearchValue(object)));
 
-        List<Subscription> subscriptionsToEmail;
+        if (artefact.getSearch().get("cases") != null) {
+            artefact.getSearch().get("cases").forEach(object -> subscriptionList.addAll(extractSearchValue(object)));
+        }
+
+        List<Subscription> subscriptionsToContact;
         if (artefact.getSensitivity().equals(Sensitivity.CLASSIFIED)) {
-            subscriptionsToEmail = validateSubscriptionPermissions(subscriptionList, artefact.getListType());
+            subscriptionsToContact = validateSubscriptionPermissions(subscriptionList, artefact.getListType());
         } else {
-            subscriptionsToEmail = subscriptionList;
+            subscriptionsToContact = subscriptionList;
         }
 
-        log.info("Subscriber list created. Notifying {} subscribers.", subscriptionsToEmail.size());
+        List<Subscription> subscriptionsForEmail = sortSubscriptionByChannel(subscriptionsToContact, Channel.EMAIL);
+
+        Map<String, List<Subscription>> returnedMappings =
```

Comment:
seems like you dont need to assign this here, you instantly use it

Reply:
and probably from here to the rest of the method, should be extracted, the method `collectSubscribers()` has collected at line 137 ish, the rest isnt to do with collecting, should trigger new handler method

Reply:
👍